src/nedap/utils/test/matchers.clj

```diff
@@ -0,0 +1,24 @@
+(ns nedap.utils.test.matchers
+  "CLJ only while in alpha, eventually CLJS could be supported by using js/Promise"
+  (:require
+   [nedap.utils.test.impl :refer [when-matcher-combinators]]))
+
+(when-matcher-combinators
+ (require 'matcher-combinators.core
+          'matcher-combinators.matchers)
+
+ (defrecord InAnyOrder [expected timeout]
+   matcher-combinators.core/Matcher
+   (match [_this actual]
+     (let [result (deref
+                   (future (matcher-combinators.core/match (matcher-combinators.matchers/in-any-order expected) actual))
+                   timeout
+                   ::timeout)]
+       (if (#{::timeout} result)
+         (throw (ex-info "in-any-order timed out", {:expected expected
+                                                    :actual   actual}))
+         result))))
+
+ (defn in-any-order [expected & {:keys [timeout]
+                                 :or {timeout 5000}}]
```

Comment:
I'd rather keep the matcher a drop-in replacement. 5sec is a _bit_ on the low side maybe. It's effectively a safeguard against extremely long loops taking forever; `in-any-order` is usually pretty fast.

We'll QA across internal projects and see what to do about the timeout

Reply:
Got it. Since using a timeout more of a workaround than a thing that is intrinsically part of the test, it seems ok.

You could add a docsring explaining these btw